### PR TITLE
Add Bors integration to repository

### LINKED ISSUE #268: Integrate Bors
Do we want to use bors with this repository in order to keep the build badge "evergreen"?

### PR DESCRIPTION
**Pull request by ebkalderon**
This pull request introduces [Bors](https://bors.tech/) build bot integration to the main Amethyst repository. Benefits of Bors include stateless CI testing and ensuring the stability of the `master` and `develop` branches. I have already installed the Bors application to our GitHub organization, so if this pull request works for us, we can adopt identical changes to all other Amethyst repositories.

Closes #268

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -1,47 +1,51 @@
...
+matrix:
+  - channel: stable
+    target: x86_64-pc-windows-msvc
+  - channel: beta
+    target: x86_64-pc-windows-msvc
+  - channel: nightly
+    target: x86_64-pc-windows-msvc
+
+# Allow for occasional compiler breakage on nightly Rust.
 matrix:
```
Comment (Xaeroxe) [inline appveyor.yml]: Error parsing appveyor.yml: (Line: 18, Col: 1, Idx: 406) - (Line: 18, Col: 7, Idx: 412): Duplicate key

Is it possible the indentation on the two lines below this were necessary?  Alternatively I think we can just move `allow_failures` into the matrix key from above.

File: appveyor.yml [hunk 2]
```diff
@@ -1,47 +1,50 @@
...
-        - channel: nightly
+  - channel: stable
+    target: x86_64-pc-windows-msvc
+  - channel: beta
+    target: x86_64-pc-windows-msvc
+  - channel: nightly
+    target: x86_64-pc-windows-msvc
+
+  # Allow for occasional compiler breakage on nightly Rust.
+  allow_failures:
```
Comment (Xaeroxe) [inline appveyor.yml]: stick a `-` here

Error parsing appveyor.yml: (Line: 18, Col: 3, Idx: 410) - (Line: 18, Col: 3, Idx: 410): While parsing a block collection, did not find expected '-' indicator.

**Comment by Xaeroxe** [comment]
On another note: I'm so glad we aren't using yml for Amethyst configs anymore.

**Comment by Xaeroxe** [comment]
Alright let's give this a shot.

@bors r+

**Comment by ebkalderon** [comment]
@Xaeroxe The merge will not work because I have placed the *status: working* label on this pull request. 😉 It is not fully ready yet, hang tight!

**Comment by Xaeroxe** [comment]
haha sorry I jumped the gun a bit.

**Comment by ebkalderon** [comment]
It's okay, it happens!

File: appveyor.yml [hunk 3]
```diff
@@ -1,47 +1,52 @@
...
+  - rustc -V
+  - cargo -V
+
+# Build the Bors staging and trying branches, plus PRs to develop and master.
+branches:
+  only:
+    - staging
+    - trying
+    - develop
+    - master
```
Comment (torkleyy) [inline appveyor.yml]: Should also build the latest release branch.

**Comment by torkleyy** [comment]
@Xaeroxe You would write "bors r+" btw.

File: appveyor.yml (hunk 3 above)
Comment (ebkalderon) [reply appveyor.yml]: Ah, good catch. Will add this now.

**Comment by Xaeroxe** [comment]
@torkleyy Wouldn't that comment count as an "approval" just now?

**Comment by torkleyy** [comment]
No, it has to be on a separate line.

**Comment by Xaeroxe** [comment]
Ah, thanks!

**Comment by Xaeroxe** [comment]
So what precisely is left to do before this should be merged @ebkalderon ?

**Comment by ebkalderon** [comment]
@Xaeroxe I've already done the extra fixes I wanted; I removed the workaround for Cargo in AppVeyor, since the HTTPS issues have already been fixed on AppVeyor's end, I've added appropriate regex to let Bors build all the `release-X.Y` branches, and finally I've documented the `bors.toml` file with comments.

Currently, I'm waiting for all the CI checks to complete. Once they're done, we can merge.

**Comment by Xaeroxe** [comment]
@ebkalderon Do you have access to the queue page for bors?  Sort of like this: https://buildbot2.rust-lang.org/homu/queue/rust I know I'm gonna get anxious in the future waiting to see if bors heard my request or not.

**Comment by Xaeroxe** [comment]
Aw :( @ebkalderon Is there any easy way to just add engine-devs as a reviewer?

**Comment by Xaeroxe** [comment]
Hey look! https://app.bors.tech/repositories/908/ If you chop off some of that URL you get a nice repository page for us!

**Comment by Xaeroxe** [comment]
bors retry

**Comment by Xaeroxe** [comment]
I guess that command doesn't work for this bors.  We should probably increase the time out before merging anyways

**Pull request merged**